php/docker/templates/nginx-vhost-conf.d/80-index.conf

```diff
@@ -1,12 +1,73 @@
 # vim: set ft=nginx:
 
+{{- $truthStrings := list "true" "on" "yes" }}
+
 location / {
     try_files $uri $uri/ /index.php$is_args$args;
 }
 
 location ~ \.php$ {
+    {{ if (has (default "off" .Env.PAGE_CACHE | lower) $truthStrings) }}
+    set $skip_cache 0;
+
+    # POST requests and urls with a query string should always go to PHP
+    if ($request_method = POST) {
+        set $skip_cache 1;
+    }
+    if ($query_string != "") {
+        set $skip_cache 1;
+    }
+
+    # Don't cache uris containing the following segments
+    if ($request_uri ~* "/wp-admin/|/xmlrpc.php|wp-.*.php|/feed/|index.php|sitemap(_index)?.xml") {
+        set $skip_cache 1;
+    }
+
+    # Don't use the cache for logged in users or recent commenters
+    if ($http_cookie ~* "comment_author|wordpress_[a-f0-9]+|wp-postpass|wordpress_no_cache|wordpress_logged_in") {
+        set $skip_cache 1;
+    }
+
+    set $key "nginx-cache:https$request_method$host$request_uri";
+    try_files $uri =404;
+
+    srcache_fetch_skip $skip_cache;
+    srcache_store_skip $skip_cache;
+
+    # https://github.com/openresty/srcache-nginx-module#srcache_response_cache_control
+    srcache_response_cache_control off;
+
+    set_escape_uri $escaped_key $key;
+
+    srcache_fetch GET /redis-fetch $key;
+    srcache_store PUT /redis-store key=$escaped_key&exptime=120;
+
+    more_set_headers 'x-cache-fetch $srcache_fetch_status';
+    more_set_headers 'x-cache-store $srcache_store_status';
+    {{ end }}
+
     fastcgi_pass   $upstream;
     fastcgi_read_timeout {{ max 60 (add 10 (default "30" .Env.PHP_REQUEST_TIMEOUT | atoi)) }};
     fastcgi_index  index.php;
     include        /usr/local/openresty/nginx/conf/fastcgi.conf;
 }
+
+{{ if (has (default "off" .Env.PAGE_CACHE | lower) $truthStrings) }}
```

Comment:
here we can use an env variable like `STACK_PAGE_CACHE_BACKEND=memcached|redis|custom` which chooses between different backend configurations.

For `redis` we generate this configuration and for `memcached` we generate correspondingly while for `custom` we don't generate any endpoints and we let the user to define its configuration.

Reply:
@bogdanpetrea what do you think ?